src/main/java/org/elasticsearch/index/mapper/MapperUtils.java

```diff
@@ -76,7 +82,50 @@ public void addConflict(String mergeFailure) {
                 throw new ElasticsearchIllegalStateException("Merging dynamic updates triggered a conflict: " + mergeFailure);
             }
         };
-        mergeInto.merge(mergeWith, ctx);
     }
 
+    /**
+     * Merge {@code mergeWith} into {@code mergeTo}. Note: this method only
+     * merges mappings, not lookup structures. Conflicts are returned as exceptions.
+     */
+    public static void merge(Mapper mergeInto, Mapper mergeWith) {
+        mergeInto.merge(mergeWith, newStrictMergeContext());
+    }
+
+    /**
+     * Merge {@code mergeWith} into {@code mergeTo}. Note: this method only
+     * merges mappings, not lookup structures. Conflicts are returned as exceptions.
+     */
+    public static void merge(Mapping mergeInto, Mapping mergeWith) {
+        mergeInto.merge(mergeWith, newStrictMergeContext());
+    }
+
+    public static void validateDynamicMappingsUpdateOnMaster(Client client, ESLogger logger, String index, String type, Mapping update) throws IOException {
+        validateDynamicMappingsUpdateOnMaster(client, logger, index, type, update, null);
+    }
+
+    /**
+     * Apply the given mapping update on the master node and throw an exception
+     * if it failed to apply.
+     */
+    public static void validateDynamicMappingsUpdateOnMaster(Client client, ESLogger logger, String index, String type, Mapping update, TimeValue timeout) throws IOException {
+        assert update != null;
+        XContentBuilder builder = XContentFactory.jsonBuilder().startObject();
+        update.toXContent(builder, new ToXContent.MapParams(ImmutableMap.<String, String>of()));
+        final String mappingUpdate = builder.endObject().string();
+        logger.debug("Validating dynamic mappings update on master node: [{}][{}]", index, mappingUpdate);
+        long start = System.nanoTime();
+        PutMappingRequestBuilder requestBuilder = client.admin().indices()
+                .preparePutMapping(index)
+                .setType(type)
+                .setSource(mappingUpdate);
+        if (timeout != null) {
+            requestBuilder.setTimeout(timeout);
+        }
+        final PutMappingResponse response = requestBuilder.get();
+        if (response.isAcknowledged() == false) {
+            throw new ElasticsearchIllegalStateException("Failed to validate dynamic mappings updates on the master node - response was not acknowledged");
```

Comment:
Is illegal state the right exception to use? I would normally use this for inconsistent state (meaning we have broken code)?